feature_engine/selection/drop_psi_features.py

```diff
@@ -387,28 +382,27 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
         if self.switch:
             test_df, basis_df = basis_df, test_df
 
-        if self.threshold == "auto":
-            threshold = self._calculate_auto_threshold(
-                basis_df.shape[0], test_df.shape[0]
-            )
-        else:
-            threshold = self.threshold
-
-        # set up the discretizer
-        if self.strategy == "equal_width":
-            bucketer = EqualWidthDiscretiser(bins=self.bins)
-        else:
-            bucketer = EqualFrequencyDiscretiser(q=self.bins)
+        # set up the discretizer for numerical variables
+        if len(self.num_variables_) > 0:
+            if self.strategy == "equal_width":
+                bucketer = EqualWidthDiscretiser(bins=self.bins)
+            else:
+                bucketer = EqualFrequencyDiscretiser(q=self.bins)
 
         # Compute the PSI by looping over the features
         self.psi_values_ = {}
         self.features_to_drop_ = []
 
         for feature in self.variables_:
```

Comment:
instead of looping in `self.variables_` and then `self.numerical_` we should just loop over numerical and then over categorical, and, like this, remove the nested looping.

Could we fix that please?